### PR TITLE
AC-1612: Upgrade react-router-dom, Log all page change events to Segment

### PR DESCRIPTION
**Pull request by gfriedmansp**
### What Changed
 - Upgraded react-router-dom to make use of `useHistory` hook
 - Created Segment helper functions to `identify` and `page`
 - On every route change (pathname change), sends a Segment `page` event.

### How To Test
 - Run 2web2ui, change pages, see events log in: https://app.segment.com/sparkpost-daniel-chalef/sources/webui_dev/debugger
 - Test other routes work correctly because of the react-router-dom upgrade

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -104,14 +104,14 @@
...
-    "react-helmet": "^5.2.0",
+    "react-helmet": "^6.1.0",
     "react-instantsearch": "^5.3.1",
     "react-load-script": "0.0.6",
     "react-loadable": "^5.3.1",
     "react-qr-svg": "^2.1.0",
     "react-recaptcha": "^2.3.8",
     "react-redux": "^5.1.2",
-    "react-router-dom": "^4.1.1",
+    "react-router-dom": "^5.2.0",
```
Comment (shrirupa) [inline package.json:114]: 🎉

File: src/components/segment/SegmentIdentify.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      });
+    }
+  }, [accessControlReady, traits]);
+
+  return null;
+};
+
+const mapStateToProps = state => ({
+  accessControlReady: state.accessControlReady,
+  [SEGMENT_TRAITS.USER_ID]: usernameSelector(state),
```
Comment (shrirupa) [inline src/components/segment/SegmentIdentify.js:23]: Aren't CID's more useful than username ? is there a reason to go for username ?

File: src/components/segment/SegmentPage.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import { segmentPage } from 'src/helpers/segment';
+
+export const SegmentPage = ({ accessControlReady }) => {
+  const history = useHistory();
+  const prevPathname = useRef(history.location.pathname);
+
+  // On URL path changes, inform segment via "PAGE" event
+  useEffect(() => {
+    if (accessControlReady) {
+      const unlisten = history.listen(location => {
```
Comment (shrirupa) [inline src/components/segment/SegmentPage.js:13]: this is nice 🤩

File: src/components/segment/SegmentIdentify.js (hunk 2 above)
Comment (gfriedmansp) [reply src/components/segment/SegmentIdentify.js:24]: 👍 I added both

**Pull request merged**